### ISSUE TITLE
damaged file

### ISSUE THREAD

**Issue by johnabuck** [agilepro/cognoscenti#434]
got this error message. Was uploading large video file and wanted to look at a topic (meeting notes). Upload stopped and this error message displayed. User in India got same message - we can reach the notes file.

![image](https://user-images.githubusercontent.com/10082365/65596620-3d05ec00-df65-11e9-9b67-c05e76c39ce7.png)

**Comment by agilepro** [comment]
at 09/25/2019 11:14:30 (GMT) John Buck started to do an upload to "broken-workspace-from-john" apparently on thread 30.   

At 09/25/2019 11:16:00 three web requests for notesList.htm were initiated and blocked waiting to get the lock.    This resulted in three exception reports.

then at 2019-09-25 11:16:28 on thread 30 following error occurred:

Unable to handle PUT to https://s06.circleweaver.com/weaver/api/socio/broken-workspace-from-john/temp/~tmp~DUWCYJKNH~tmp~?lic=YOUANHQMH%21ZOUANHQMH"

Unexpected EOF read on the socket

What his means is that John was trying to upload a file, but the transfer was cut off.  The socket disappeared before the expected end.

at 019-09-25 11:19:09 this error
failed after 10 seconds to set a lock for container (broken-workspace-from-john), Lock held by tid=28

**Comment by agilepro** [comment]
I am convinced that for some reason, after 2 minutes of uploading, something cut off the connection.

I found one problem in the code.   It had been grabbing a workspace lock for the entire upload.  That means that while you are uploading, the workspace was locking everyone else out.

In this example people were locked for two minutes at least, and maybe beyond.   I have managed to release the lock BEFORE receiving the file.   This proved to be able to allow multiple files to upload at the same time (overlapping).  This was not possible before, and also there should be less problem with other requests.
